cartoframes/context.py

```diff
@@ -244,11 +246,33 @@ def read(self, table_name, limit=None, decode_geom=False, shared_user=None):
 
         query = 'COPY {source} TO stdout WITH (FORMAT csv, HEADER true)'.format(source=source)
 
-        result = self.copy_client.copyto_stream(query)
+        result = self._recursive_read(query, retry_times)
         df = pd.read_csv(result)
 
         return self._clean_DataFrame_from_carto(df, table_name, schema, decode_geom)
 
+    def _recursive_read(self, query, retry_times=3):
```

Comment:
Correct me if I'm wrong, but the way I read this is that there would be `n-1` re-tries instead of `n` re-tries. That is, there would be `n` tries, not `n` re-tries.

Reply:
:+1: